src/proxy/routes-engine.ts

```diff
@@ -0,0 +1,90 @@
+import fs from "fs";
+import http from "http";
+import { DEFAULT_CONFIG } from "../config";
+import { decodeCookie, findFile } from "../core/utils";
+
+export const handleUserCustomRoutes = async (folder: string): Promise<UserDefinedRoute[]> => {
+  if (!fs.existsSync(folder)) {
+    return [];
+  }
+
+  const routesFile = await findFile(folder, DEFAULT_CONFIG.swaConfigFilePattern!);
+  if (!routesFile) {
+    return [];
+  }
+
+  let routes = [];
+  try {
+    routes = require(routesFile).routes || [];
+  } catch (error) {}
+
+  if (routes.length > 0) {
+    console.log("reading routes definition", routesFile);
+    return routes;
+  }
+  return routes;
+};
+
+export const processUserRoute = async (req: http.IncomingMessage, res: http.ServerResponse, userDefinedRoutes: UserDefinedRoute[]) => {
+  if (!req) {
+    return Promise.resolve(-1);
+  }
+
+  const userDefinedRoute = userDefinedRoutes.find((routeDef) => new RegExp(routeDef.route).test(req.url!));
+  console.log({ userDefinedRoute });
+
+  if (userDefinedRoute) {
+    // set headers
+    if (userDefinedRoute.headers) {
+      for (const header in userDefinedRoute.headers) {
+        res.setHeader(header, userDefinedRoute.headers[header]);
+      }
+    }
+
+    // check allowed method
```

Comment:
Method is part of the matching criteria. If the url path matches but there are methods defined, and no methods match, the route is not a match.